siriuspy/siriuspy/devices/ids.py

```diff
@@ -87,22 +133,113 @@ def idkx(self, value):
         """Set APU Kx."""
         self['Kx-SP'] = value
 
+    # --- movement checks ---
+
     @property
     def is_moving(self):
         """Return True if phase is changing."""
         return round(self['Moving-Mon']) == 1
 
-    def cmd_move(self, timeout=_DEF_TIMEOUT):
-        """."""
-        self['DevCtrl-Cmd'] = APU._CMD_MOVE
-        return True
+    # --- cmd_beamline and cmd_drive
+
+    def cmd_beamline_ctrl_enable(self, timeout=None):
+        """Command enable bealine ID control."""
+        return self._write_sp('BeamLineCtrlEnbl-Sel', 1, timeout)
+
+    def cmd_beamline_ctrl_disable(self, timeout=None):
+        """Command disable bealine ID control."""
+        return self._write_sp('BeamLineCtrlEnbl-Sel', 0, timeout)
+
+    # --- cmd_set ---
+
+    def cmd_set_phase(self, phase, timeout=None):
+        """Command to set ID target phase for movement [mm]."""
+        return self._write_sp('Phase-SP', phase, timeout)
+
+    def cmd_set_phase_speed(self, phase_speed, timeout=None):
+        """Command to set ID cruise phase speed for movement [mm/s]."""
+        return self._write_sp('PhaseSpeed-SP', phase_speed, timeout)
+
+    def cmd_set_phase_speed_max(self, phase_speed_max, timeout=None):
+        """Command to set ID max cruise phase speed for movement [mm/s]."""
+        return self._write_sp('MaxPhaseSpeed-SP', phase_speed_max, timeout)
+
+    # --- cmd_wait
 
     def wait_move(self):
         """Wait for phase movement to complete."""
         _time.sleep(APU._MOVECHECK_SLEEP)
         while self.is_moving:
             _time.sleep(APU._MOVECHECK_SLEEP)
 
+    # -- cmd_move
+
+    def cmd_move_stop(self, timeout=_default_timeout):
+        """."""
+        self['DevCtrl-Cmd'] = APU._CMD_MOVE_STOP
+        return True
+
+    def cmd_move_start(self, timeout=_default_timeout):
+        """."""
+        self['DevCtrl-Cmd'] = APU._CMD_MOVE_START
+        return True
+
+    def cmd_move(self, phase, timeout=None):
```

Comment:
Once this method has an argument `phase`, I think a more suitable name would be something with a `set` prefix, like `set_phase` or `set_wait_phase`... What do you think?

Reply:
changed